File: 백준/Silver/2485. 가로수/가로수.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int t = Integer.parseInt(br.readLine());
        int arr[] = new int [t];

        for (int i = 0; i < t; i++) {
            arr[i] = Integer.parseInt(br.readLine());
        }

        int gcd = 0;
        for (int i = 1; i < t; i++) {
            int temp = arr[i] - arr[i - 1];
            gcd = findGcd(temp, gcd);
        }

        int count = 0;
        for (int i = 0; i < t - 1; i++) {
            if (arr[i] + gcd == arr[i + 1]) {
                continue;
            } else {
                int num = arr[i];
                while (true) {
                    if (num + gcd == arr[i + 1]) {
                        break;
                    }
                    num += gcd;
                    count++;
                }
            }
        }

        StringBuilder sb = new StringBuilder();
        sb.append(count);

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }

    private static int findGcd(int a, int b) {
        if (b == 0) {
            return a;
        } else {
            return findGcd(b, a % b);
        }
    }
}

File: 백준/Silver/2485. 가로수/가로수.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val t = br.readLine().toInt()
    val arr = mutableListOf<Int>()
    repeat(t) {
        arr.add(br.readLine().toInt())
    }

    var gcd = arr[1] - arr[0]
    for (i in 2 until t) {
        val diff = arr[i] - arr[i - 1]
        gcd = findGcd(gcd, diff)
    }

    var count = 0
    for (i in 0 until t - 1) {
        if (arr[i] + gcd == arr[i + 1]) {
            continue
        } else {
            var num = arr[i]
            while (true) {
                if (num + gcd == arr[i + 1]) {
                    break
                }
                num += gcd
                count++
            }
        }
    }
    val sb = StringBuilder()
    sb.append(count)

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}

fun findGcd(a: Int, b: Int): Int {
    if (b == 0) {
        return a
    } else {
        return findGcd(b, a % b)
    }
}
